### ISSUE TITLE
[java] UseEqualsToCompareStrings false positive with PMD 6.30.0

### ISSUE THREAD

**Issue by matthiaskraaz** [pmd/pmd#3004]
Rule: errorprone.xml/UseEqualsToCompareStrings

The following code generates a false positive:

```
  boolean f(String s) {
    return s.charAt(0) == s.charAt(1);
  }
```

**Comment by oowekyala** [comment]
Possibly related to #2976, #2920
